components/RefinementList.js

```diff
@@ -43,6 +45,7 @@ class RefinementList extends React.Component {
     var rootClass = cx(this.props.cssClasses.root);
     var listClass = cx(this.props.cssClasses.list);
     var itemClass = cx(this.props.cssClasses.item);
+    var transformData = this.props.transformData || identity;
 
```

Comment:
> There is no performance problem involve here

100% agree

> and it makes the code easier to grasp (no if)

This is true only if you know what is lodash.identity. The first time I encountered this kind of code I wondered "What is identity doing?". Then I had to go to the lodash documentation and find identity documentation only to understand that it was doing nothing at all (a noop with a return).

I rarely use noop() and identity(), only in cases for example an API forces me to pass a parameter while I do not have the usage of it (eg a library force passing a listener that I will not use).

Here we do control the code and still, we are executing a function while we know there's nothing to do at all. So yes it seems to make the implementation code smarter but in the end it's a bit lying, a simple if would make the code exhaustive without having to use and understand lodash.identity.

This is really a detail and not linked to the awesomeness of the feature in the PR but I wanted to explain the reasoning here.

This do not restrict us to just merge this and move on.